### DISCUSSION TITLE
Energy usage / Water usage

### DISCUSSION THREAD

**Discussion by davidrwalton** [ekutner/home-connect-hass#451] [General]
Not sure if this should be a feature request.  My Bosch Dishwasher seems to report energy and water usage as a percentage in HA not sure I understand why it would be that as theirs nothing in the doc for this plugin. Could this be fixed at some point or is this a limitation of what’s in the api? I’m sure if it was available it would be a great addition.

cheers

**Comment by jensrossbach** [comment]
I agree, energy and water consumption as percentage is pretty useless (percent of what?). I have the same issue. On the other side, the app reports energy and water consumption in kWh and liters. Can this be fixed?

**Comment by ekutner** [comment]
This is what's coming from the API, you can override it if you really wanted to, but what would be the meaning of changing the unit when the data arrives as percentage numbers?

**Comment by jensrossbach** [comment]
Is that part of the API documented somewhere? Maybe there's some explanation what the percentage values refer to.

**Comment by ekutner** [comment]
I didn't find any documentation for it. I guess it's a percent of the maximum cycle consumption, whatever that may be.

**Comment by Cryonic90** [comment]
Is it possible to also report this for a cooktop? I can see it on the device itself, but it is not shown in the integration. Not sure if the API reports the energy consumption for a cooktop?

**Comment by DemianTinkiel** [comment]
Sounds like these are percentage of average consumption.

You could use a template to do this, with values based on the max liter and wattage specs of your device, or if you're lazy like me, just use what you see in the home connect app. In my case
```
Energy : 79% × 1.5189873417722 = 1.2
Water: 56% × 20.535714285714 = 11.5
```
E.g
```
{{
  states("sensor.012030389287015391_bsh_common_option_waterforecast")|float/100*20.535714285714|round(1)
}}
```
I just put this in a mushroom chip template card but making your own custom sensor is probably better for reusability 
![Screenshot_20241110_170124_Home Assistant](https://github.com/user-attachments/assets/73e2445d-631f-448a-8199-e8e1a68423b0)

Note that
1. This is statistical (they say the same in the app)
2. I must be doing something wrong with the round function at the end but you get the point 😅

**Comment by ebioman** [comment]
@DemianTinkiel where did you modify this, I am struggling to find the right yaml and entry for this, or did you only to that for the widget?

**Comment by DemianTinkiel** [comment]
> @DemianTinkiel where did you modify this, I am struggling to find the right yaml and entry for this, or did you only to that for the widget?

@ebioman I only did it for that widget. in my case I  did it on a dashboard card with a mushroom-chip e.g.
 ```
chips:
      - type: template
        content: |-
          {{
            states("sensor.012030389287015391_bsh_common_option_waterforecast")|float/100*20.535714285714|round(1)

          }} L
```
I just did a quick check on how to do it as a sensor: settings > devices and services > helpers > create helper. From there select template > template a sensor 
![Screenshot_20241125_073948_Chrome](https://github.com/user-attachments/assets/7e724128-92f4-4478-9424-4c0f7c3cf409)

**Comment by ebioman** [comment]
thx, giving it a try

**Comment by helpfinder** [comment]
Hi all,
I tried this, but should this work dynamically?
because I can create Helper like
states("sensor.012030389287015391_bsh_common_option_waterforecast")|float/100*20.535714285714|round(1)

with calculated value for multiplication but this will be still the same L or kWh number, won´t?
At least for me I created helper, get a sensor which was the same value in L and kWh as in Bosch app, but after next dishwasher cycle mobile app increased consumed total energy and water, but obviously Helper is still on the same L, kWh value

Energy and Water in mobile app does not represent current/actual energy/water consumption, but cumulative

@DemianTinkiel what is your use case in HA?

Many thanks

**Comment by iceymanz** [comment]
I've come to the conclusion that there is no meaningful water or power consumption data to be extracted from these percentages. They're a super simplified way of showing you which of the obscurely named options use more power or water rather than less. In my washing machine, at least, the device will vary its draw based on various sensors during the cycle in a way the percentages can't account for. For anyone moving forward may I suggest: 1. put a quality power sensing plug on your device to capture the actual real energy usage straight into HA's energy monitoring. 2. If water draw is important there are sensors but I'm not across those.

**Reply by Greentreegarden** [reply]
Totally agree to @iceymanz - "no meaningful power or water consumption".

I put a BlitzWolf BW-SHP13 smart plug to measure power consumption. I found 0,9 kWh per 45-60°C washing program.
I attached the dishwasher to the hot water tap, where the heat pump generates the 44°C/55°C hot water. Consumption dropped to 0,7 kWh.

We live off-grid. Any kWh matters on overcast days - and heating water is the worst!

I did not have a water flow meter, and we have water in abundance, so water consumption doesn't matter.

**Comment by DemianTinkiel** [comment]
@helpfinder 
> I tried this, but should this work dynamically?

It does for me. If I change a program or an option my sensor changes 
![Screenshot_20241207_084203_Home Assistant](https://github.com/user-attachments/assets/ac49b699-281a-4138-9484-0b6d344d7d02)

> Energy and Water in mobile app does not represent current/actual energy/water consumption, but cumulative

You're right that is not real consumption, they say that themselves in the app, it's just an estimate based on what program you have selected etc
![Screenshot_20241207_083958_Home Connect](https://github.com/user-attachments/assets/fdbf7136-831a-4a73-85ac-91e5a06c55f4)

But in terms of being cumulative, that depends on how you set it up in HA. I'd say have a look at the docs on the [state class](https://developers.home-assistant.io/docs/core/entity/sensor/#available-state-classes)

![Screenshot_20241207_211821_Chrome](https://github.com/user-attachments/assets/c49d5aaf-6955-46f8-b186-bf0da7ef90c5)

**Comment by helpfinder** [comment]
@DemianTinkiel Thank you for reply
Then I did my Helper probably somehow wrong - wrong formula. Can you please share more details how to do so?

I creatd it like this
![image](https://github.com/user-attachments/assets/db075476-eb52-4b3c-b530-a28721c873e1)

but in Home connect mobile app I see 7,2 kWh as consumption (the same for water)

btw any clue how to change gallons to liters in the mobile app? I did not find the setting anywhere, I also send email to Bosch support regarding this, but they are ignoring me :)

**Comment by DemianTinkiel** [comment]
@helpfinder 
Formula looks good to me. The magic number there is the main difference. To be sure, check it with a calculator like 
https://www.calculator.net/percent-calculator.html
This should be the 100% of what ever you're trying to get the sensor for. I.e. `value_in_app / value_in_built-in_sensor% = x` 
For example assuming water usage (see my 1st post): `11.5/56%=20.5357...` 

Then I'd check if the built in sensor (the one that comes with the integration) updates every time you change the program.  Depending on the server you connect to there might be a delay between changing the dishwasher program and seeing a new estimate

> btw any clue how to change gallons to liters in the mobile app? 

 No idea, sorry!

**Comment by ekutner** [comment]
FYI, this has become a philosophical discussion so I'll be moving it to the discussions section.

**Comment by ebioman** [comment]
Thanks the following did it for me to get liters:

'''
{{
states("sensor.404010386856010541_bsh_common_option_waterforecast")|float/100*5*3.785411784|round(1)
}}
'''

**Comment by barndawgie** [comment]
These posts above all seem to give a way of estimating the cost per cycle using that perecentage field that comes back.

How are you all turning that into an actual sensor? Do you have an automation that increments a counter each time the state transition to Finished? Some other simpler thing that I'm not thinking of?
